kinode/src/net/mod.rs

```diff
@@ -322,20 +332,27 @@ async fn handle_local_request(
     }
 }
 
-async fn handle_fdman(km: &KernelMessage, request_body: &[u8], data: &NetData) {
+async fn handle_fdman(km: &KernelMessage, request_body: &[u8], data: &mut NetData) {
     if km.source.process != *lib::core::FD_MANAGER_PROCESS_ID {
         return;
     }
     let Ok(req) = serde_json::from_slice::<lib::core::FdManagerRequest>(request_body) else {
         return;
     };
     match req {
-        lib::core::FdManagerRequest::Cull {
-            cull_fraction_denominator,
-        } => {
-            // we are requested to cull a fraction of our peers!
+        lib::core::FdManagerRequest::FdsLimit(fds_limit) => {
+            data.fds_limit = fds_limit;
+            if data.max_peers > fds_limit {
+                data.max_peers = fds_limit;
+            }
+            // TODO combine with max_peers check
+            if data.max_passthroughs > fds_limit {
+                data.max_passthroughs = fds_limit;
+            }
             // TODO cull passthroughs too?
-            data.peers.cull(cull_fraction_denominator).await;
+            if data.peers.peers().len() >= data.fds_limit as usize {
+                data.peers.cull(2).await;
```

Comment:
this doesn't seem quite right: seems like cull needs to be based on the difference to guarantee we are within the limit?